### PR TITLE
Bulk import validation - species

### PR DESCRIPTION
**Pull request by holmbergius**
This PR addresses missing or unsupported species columns (Encounter.genus and Encounter.specificEpithet) in bulk import spreadsheets. However, it lays the foundation for reporting errors in other fields as well and blocking submission where data is missing.

Specifically:

- Adds method RowFeed.logParseError(int colNum, Object value, Row row, String exactMessageToDisplay) for more informative error messages via parameter exactMessageToDisplay
- Adds an Errors section to the Import Overview preview
- Provides a count of total errors in the spreadsheet, which can include missing images and incorrect species currently
- Throws an error if Encounter.genus and/or Encounter.specificEpithet are blank or empty strings
- Throws an error if Encounter.genus and Encounter.specificEpithet form a taxonomy not defined in commonConfiguration.properties
- Lists columns with errors in the Errors summary to help focus the user
- Blocks the "Commit these results." button if errors are >0, allowing for future behavior to also add more errors to the list of blocking errors.

image

### CODE REVIEW

File: src/main/java/org/ecocean/servlet/importer/StandardImport.java [hunk 1]
```diff
@@ -894,10 +931,40 @@ public Encounter loadEncounter(Row row, ArrayList<Annotation> annotations, Strin
...
+    	feedback.logParseError(getColIndexFromColName("Encounter.genus", colIndexMap),"GENUS", row, "MISSING GENUS");
+    }
+    if (!hasSpecificEpithet) {
+        //mark specific epithet
+        feedback.logParseError(getColIndexFromColName("Encounter.specificEpithet", colIndexMap),"SPECIFIC EPITHET", row, "MISSING SPECIFIC EPITHET");
+    }
+    //now validate that a present genus and species value are supported
+    if(hasGenus && hasSpecificEpithet) {
+    	
+    	List<String> configuredSpecies = CommonConfiguration.getIndexedPropertyValues("genusSpecies", myShepherd.getContext());
```
Comment (naknomum) [inline src/main/java/org/ecocean/servlet/importer/StandardImport.java:959]: how is this _not_ already a method we have somewhere? :exploding_head: (i looked, and cant seem to find it [either]. closest might be `Shepherd.getAllTaxonomyNames()`.) i wonder if we should make this a method we can use in a hopeful future - maybe inside Taxonomy class?

Comment (naknomum) [reply src/main/java/org/ecocean/servlet/importer/StandardImport.java:959]: i meant the above line _plus_ the one below: basically **validating** a genus/specificEpithet pair.

**Review by naknomum — APPROVED** [review]
looks great to me. glad it sets up a way to handle other errors in the future.
made one comment, more of a "bonus" flavor; so going to go ahead and approve either way.

File: src/main/java/org/ecocean/servlet/importer/StandardImport.java (hunk 1 above)
Comment (holmbergius) [reply src/main/java/org/ecocean/servlet/importer/StandardImport.java:959]: Good idea. I will implement this in the next pass at adding checks.

**Comment by T4Cmaureen** [comment]
Something else that could use fixing in the bulk import validation is the fact that the "Not Imported column types" flags submitter and/or photographer account information when the information is correct in the user's bulk import spreadsheet. These columns will also correspondingly be highlighted in yellow in the Parsed Import Table. It also doesn't flag these columns consistently - sometimes it might flag the submitter email column or the submitter full name, but on a subsequent upload it might list & highlight different user or photographer account fields, all while the user's spreadsheet is listing them correctly. I have to explain to everyone that what it's likely doing is trying to indicate that that account already exists in the system. But it's a false flag and can lead to users believing they should remove this info from their spreadsheets which then leads to the data not being assigned to their account. It would be very useful to remove or otherwise re-design this misleading validation action for these columns/fields. thanks

**Pull request merged**